plugins/modules/ipaautomountmap.py

```diff
@@ -124,9 +127,10 @@ def check_ipa_params(self):
 
     def get_args(self, mapname, desc):  # pylint: disable=no-self-use
         _args = {}
-        if mapname:
-            _args["automountmapname"] = mapname
-        if desc:
+        if not mapname:
+            raise ValueError("Automount map name is required.")
```

Comment:
Why is this needed here? A check to have valid names should be done early before any IPA command is used.

Reply:
This is a somewhat paranoid test, but it might happen. Consider this playbook task:
```yaml
- ipaautomountmap:
    name:
      - ''
    description: "an empty name map"
```
This task will pass validation (`len(name) == 1`) on `automountmap` and on (maybe) all other modules, but the name itself is invalid.

Should we test for empty names (inside `for name in names`) or should we rely on IPA API checks?

If we check, there will be less API calls and we can fail early, if we rely on IPA API, we'll have one less error check to deal on modules, but we might get less descriptive errors (I haven't look at the possible errors, I'm just speculating.)

Reply:
The issue also might happen on `state: absent`:
```yaml
ipaautomountmap:
  name:
     - test_map
     - ''
     - another_map
   state: absent
```

Reply:
I think we should check this before doing anything with the API.

Reply:
So... I need we need to fix it on all modules. For `ipagroup`:
```yaml
  - ipagroup:
      name:
        - ""
      state: present
```
This is the result:
```
TASK [ipagroup] ****************************************************************************************
fatal: [fedsrv.ipa.test]: FAILED! => {"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python3"}, "changed": false, "msg": "There is more than one group ''"}
```
This happens as `ipagroup` still uses `*_find` instead of `*_show`.

Reply:
We could add a check to module_params_get. Is there a use case to return a list of strings, with one or more empty strings?

Reply:
Please have a look at https://github.com/freeipa/ansible-freeipa/pull/779

Reply:
On a fast, first look, #779 looks good (I'm reviewing it). I don't remember of any case were an empty string should be allowed in a list.

Reply:
We have one place: userauthtype in the user module.

Reply:
PR #779 has beem merged. Marking this conversation as resolved.